feeder/util/mqtt/client.py

```diff
@@ -226,9 +232,7 @@ async def send_cmd_schedule(
             await self.send_cmd(gateway_id, device_id, "schedule_mod_end", {})
 
     async def start(self):
-        await self.connect(
-            "mqtt://%s:%s@localhost:1883/" % (local_username, local_password)
-        )
+        await self.connect(self._mqtt_url)
```

Comment:
I feel like all the logic that you put in the initializer would make much more sense here. It seems weird to me that we pre-collect all of the parameters when they could just be passed in here.